pkg/identity/gitlabcom/principal_test.go

```diff
@@ -173,6 +179,48 @@ func TestJobPrincipalFromIDToken(t *testing.T) {
 				runnerID:          1,
 				runnerEnvironment: "gitlab-hosted",
 				sha:               "714a629c0b401fdce83e847fc9589983fc6f46bc",
+				projectVisibility: "public",
+			},
+			WantErr: false,
+		},
+		`Token missing project_visibility is ok`: {
```

Comment:
@marshall007 want to check my thinking here, I'm currently only setting this extension for GitLab if the claim exists, and once the [gitlab MR](https://gitlab.com/gitlab-org/gitlab/-/merge_requests/125787/diffs) lands we can change this to fail, does this make sense?